src/services/goToDefinition.ts

```diff
@@ -510,4 +522,26 @@ namespace ts.GoToDefinition {
                 return false;
         }
     }
+
+    interface StatementWithExpression extends Statement {
+        readonly expression: Expression
+    }
+
+    function createDefinitionInfoFromStatement<K extends KeywordSyntaxKind>(statement: StatementWithExpression, keywordKind: K, name: string): DefinitionInfo {
+        const sourceFile = statement.getSourceFile();

```

Comment:
The caller of `createDefinitionInfoFromStatement` already has a `sourceFile`. Also, `getChildren` implicitly gets the `sourceFile` of a node.

I think you should pass along the `sourceFile` to `createDefinitionInfoFromStatement`, and then pass it through to `statement.getChildren()` as well.

Reply:
fixed